src/Mediator.Switch.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -56,12 +71,11 @@ public static IServiceCollection AddScoped<TSwitchMediator>(this IServiceCollect
         public static IServiceCollection OrderNotificationHandlers<TNotification>(this IServiceCollection services, params Type[] handlerTypes)
             where TNotification : INotification
         {
-            services.AddScoped<IEnumerable<INotificationHandler<TNotification>>>(sp =>
-            {
-                return handlerTypes
-                    .Select(handlerType => (INotificationHandler<TNotification>)sp.GetRequiredService(handlerType))
-                    .ToList();
-            });
+            services.Add(new ServiceDescriptor(typeof(IEnumerable<INotificationHandler<TNotification>>),
+                sp => handlerTypes.Select(handlerType
+                    => (INotificationHandler<TNotification>)sp.GetRequiredService(handlerType)).ToList(),
+                ServiceLifetime.Scoped));
```

Comment:
Not sure how to resolve this particular lifetime since options are not provided.  Possibly just resolve the mediator's lifetime?